### PR TITLE
add tests and fix pipeline to make it work on executors

### PR DESCRIPTION
**Pull request by nicholas-geofroy**

### CODE REVIEW

File: src/pipeline.py [hunk 1]
```diff
@@ -31,32 +38,54 @@ def run(self):
...
+        PipelineWorker(context, name=f"worker-{i}")
         for i in range(0, context['n_workers'])
     ]
 
     # Start all workers
     for worker in workers:
         worker.start()
 
+    iterations = context['max_iterations']
+    iterations = 10000 if iterations is None else iterations
```
Comment (Ercadio) [inline src/pipeline.py:71]:
```suggestion
    iterations = context.get('max_iterations', 10000)
```

File: src/pipeline.py [hunk 2]
```diff
@@ -31,32 +38,54 @@ def run(self):
...
+
+    for i in range(iterations):
+        idx = 0
+        work_queue.put(Work(executors[idx], None))
+        idx += 1
```
Comment (Ercadio) [inline src/pipeline.py:76]: This puts the same executor on the work queue
```suggestion
    for executor in executors:
        work_queue.put(Work(executor, None))
```

File: src/pipeline.py [hunk 3]
```diff
@@ -31,32 +38,54 @@ def run(self):
...
-        PipelineWorker(context, name=f"worker-{hostname}-{i}")
+        PipelineWorker(context, name=f"worker-{i}")
         for i in range(0, context['n_workers'])
     ]
 
     # Start all workers
     for worker in workers:
         worker.start()
 
+    iterations = context['max_iterations']
```
Comment (Ercadio) [inline src/pipeline.py:70]: What is iterations?

File: src/pipeline.py (hunk 2 above)
Comment (nicholas-geofroy) [reply src/pipeline.py:75]: ya that's my bad, what I meant to do was alternate between all of the ones in the list, iterations times (that way the code can end for testing at least)

**Pull request merged**